modules/cdp-engine/src/lib.rs

```diff
@@ -648,23 +648,29 @@ impl<T: Config> Pallet<T> {
 		// get the max iterationns config
 		let max_iterations = StorageValueRef::persistent(OFFCHAIN_WORKER_MAX_ITERATIONS)
 			.get::<u32>()
-			.unwrap_or(Some(DEFAULT_MAX_ITERATIONS))
-			.ok_or(OffchainErr::OffchainStore)?;
+			.unwrap_or(Some(DEFAULT_MAX_ITERATIONS));
```

Comment:
Mistake 2: When max iterations is stored as None this is not an error, just means iterator will run to completion no matter the size of the iterator. This is probably not what collators would want to set but is still a valid state

Reply:
I think we want to make None to be `DEFAULT_MAX_ITERATIONS`

Reply:
Yeah that makes sense, It is kinda a foot-gun unnecessarily for node operators